### PR TITLE
[AV-43732] Search Developer Guide for Server 7.2

### PR DESCRIPTION
**Pull request by sarahlwelton**
The long awaited PR for Search. 

Please review the content on the staging site, it's much easier than trying to review all of the raw content: 

https://docs-staging.couchbase.com/server/7.2/search/search.html

### CODE REVIEW

File: modules/search/examples/autocomplete-sample.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+using Couchbase;
+using Couchbase.Search.Queries.Simple;
+
+await using var cluster = await Cluster.ConnectAsync(new ClusterOptions
+{
+    ConnectionString = "CB_HOSTNAME", //
```
Comment (osfameron) [inline modules/search/examples/autocomplete-sample.cs]: trailing empty comment

File: modules/search/examples/autocomplete-sample.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      result.rows.forEach((row) => { console.log(row.id,row.fields)})
+    } catch (error) {
+      console.error(error)
+    }
+  }
+
+  const inputStr = "Beaufort Hotel"
+  for (let i = 2; i <= 8; i++) {
+    var testStr = inputStr.substring(0, i);
+    console.log("Input <" + testStr + ">, len: " + testStr.length)
```
Comment (osfameron) [inline modules/search/examples/autocomplete-sample.js:26]: should probably use the backtick syntax to neaten it up

File: modules/search/examples/create-search-index-header.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+curl -s -XPUT -H "Content-Type: application/json" \
+-u $CB_USERNAME:$CB_PASSWORD http://$CB_HOSTNAME:8094/api/index/landmark-content-index
```
Comment (osfameron) [inline modules/search/examples/create-search-index-header.sh]: indentation

File: modules/search/examples/create-search-index-payload.sh [hunk 4]
```diff
@@ -0,0 +1,64 @@
+curl -s -XPUT -H "Content-Type: application/json" \
+-u $CB_USERNAME:$CB_PASSWORD http://$CB_HOSTNAME:8094/api/index/landmark-content-index
```
Comment (osfameron) [inline modules/search/examples/create-search-index-payload.sh]: indentation

File: modules/search/examples/run-search-full-request.jsonc [hunk 5]
```diff
@@ -0,0 +1,113 @@
+// tag::full[]
+{
+    "query": {},
+
+    // tag::ctl[]
```
Comment (osfameron) [inline modules/search/examples/run-search-full-request.jsonc:5]: Hmm, if this kind of thing needed to be done regularly we could think about something like `include-jq::foo.json[.ctl.consistency]` to apply a jq path expression instead of having to manually tag these things 🤔

File: modules/search/examples/run-search-payload.sh [hunk 6]
```diff
@@ -0,0 +1,14 @@
+curl -XPOST -H "Content-Type: application/json" \
+-u $CB_USERNAME:$CB_PASSWORD http://$CB_HOSTNAME:8094/api/index/travel-sample.inventory.landmark-content-index/query \
```
Comment (osfameron) [inline modules/search/examples/run-search-payload.sh]: indentation.

File: modules/search/pages/create-child-field.adoc [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+. Go to *Search*.
+. Click the index where you want to create a child field.
+. Click btn:[Edit].
+. Expand *Customize Index*. 
+. Expand *Mappings*. 
+. Point to an existing mapping and click btn:[+].
+. Click *insert child field*. 
+. In the *Field* field, enter the name of a field in your documents that you want to add or remove from the index.
++
+NOTE: The field must contain either a single value or an array. If the field contains a JSON object, xref:create-child-mapping.adoc[], instead.
```
Comment (osfameron) [inline modules/search/pages/create-child-field.adoc]: missing "see" before xref?

File: modules/search/pages/create-child-field.adoc [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+
+* You've logged in to the Couchbase Server Web Console. 
+ 
+
+== Procedure 
+
+To add or remove a child field from a Search index with the Couchbase Server Web Console:
+
+. Go to *Search*.
+. Click the index where you want to create a child field.
```
Comment (osfameron) [inline modules/search/pages/create-child-field.adoc:24]: "Choose" might work here

File: modules/search/pages/create-child-field.adoc [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+. Click btn:[Edit].
+. Expand *Customize Index*. 
+. Expand *Mappings*. 
+. Point to an existing mapping and click btn:[+].
+. Click *insert child field*. 
+. In the *Field* field, enter the name of a field in your documents that you want to add or remove from the index.
```
Comment (osfameron) [inline modules/search/pages/create-child-field.adoc:29]: This feels like a long and very UI-procedural list, which means that every step seems to be as important, whether it's clickety clickety clickety click navigation, or thinking about *what* you want to change.

I wonder if we can chunk it a bit more high-level?

e.g. could we "Edit - Customize Index - Mappings" as if it was a menu selection?

or "Choose the existing mapping you want to edit. Click "+" then "insert child field".

File: modules/search/pages/create-child-field.adoc [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+. In the *Type* list, select the data type for the field: 
++
+For more information about the available data types, see xref:field-data-types-reference.adoc[].
+
+. Configure optional settings for the child field: 
++
+|====
+|Option |Description 
+
+|Searchable As a|
```
Comment (osfameron) [inline modules/search/pages/create-child-field.adoc:41]: "Searchable As a" looks weird.
Move the `a|` to a new line?

(Though actually "Searchable As" still looks weird? What does it mean?)

File: modules/search/pages/create-child-field.adoc [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+
+Set a different name that you can use to search the field's contents in a query. 
+
+The default value is the value set in *Field*.
+
+|Analyzer a|
```
Comment (osfameron) [inline modules/search/pages/create-child-field.adoc:47]: Move `a|` to new line

File: modules/search/pages/create-child-field.adoc [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+
+Choose an analyzer for the content in the child field. 
+
+You can choose a default analyzer or a custom analyzer. 
+
+For more information about the available default analyzers, see xref:default-analyzers-reference.adoc[].
```
Comment (osfameron) [inline modules/search/pages/create-child-field.adoc:53]: Can't these xrefs be inlines into line 52?

File: modules/search/pages/create-child-field.adoc [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+
+For more information about how to create a custom analyzer, see xref:create-custom-analyzer.adoc[].
+
+|Index a|
```
Comment (osfameron) [inline modules/search/pages/create-child-field.adoc:57]: Sorry, I really hate these `a|` notation, Asciidoc has some weird design decisions.
I'll stop mentioning every one now...

File: modules/search/pages/create-child-field.adoc [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+To remove the child field from the index, clear *Index*. 
+
+|[[store]]Store a|
+
+To store the values from the field in the index and return them in search results, select *Store*.
+This increases your index's size and indexing time.  
+
+To remove the field's values from the index, clear *Store*.
+
+|Include in _all field a|
```
Comment (osfameron) [inline modules/search/pages/create-child-field.adoc:70]: Backticks for `_all`

File: modules/search/pages/create-child-mapping.adoc [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+
+== Prerequisites 
+
+* You've created an index.
+For more information, see xref:create-search-index-ui.adoc[].
+
+* You've created a type mapping. 
+For more information about how to create a type mapping on an index, see xref:create-type-mapping.adoc[].
+
+* You've logged in to the Couchbase Server Web Console.
```
Comment (osfameron) [inline modules/search/pages/create-child-mapping.adoc:16]: "You are" ?

File: modules/search/pages/create-child-mapping.adoc [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+To create a child mapping with the Couchbase Server Web Console: 
+
+. Go to *Search*.
+. Click the index where you want to create a child mapping.
+. Click btn:[Edit].
+. Expand *Customize Index*. 
+. Expand *Mappings*. 
+. Point to an existing mapping and click btn:[+].
+. Click *insert child mapping*. 
+. In the *{}* field, enter the name of a field in a document that contains a JSON object.
```
Comment (osfameron) [inline modules/search/pages/create-child-mapping.adoc:28]: The field is called `{}` ?
Does asciidoc not mangle this?

File: modules/search/pages/create-custom-date-time-parser.adoc [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+. Click btn:[Edit].
+. Expand *Customize Index*. 
+. Expand *Date/Time Parsers*. 
+. Click btn:[Add Date/Time Parser]
+. In the *Name* field, enter a name for the date/time parser. 
+. In the *Layout to be added* field, enter a date/time layout with Go syntax. 
++
+For more information, see the documentation about the https://pkg.go.dev/time#pkg-constants[Go Programming Language Time Package's Layout Constant^].
+. Click btn:[Add]. 
+. (Optional) To add an additional layout, repeat Steps 7-8.
```
Comment (osfameron) [inline modules/search/pages/create-custom-date-time-parser.adoc]: Hardcoding `7-8` feels fragile.
Could you just say "add additional layouts as needed" as they already know the procedure for this now?

File: modules/search/pages/create-custom-token-filter.adoc [hunk 18]
```diff
@@ -0,0 +1,194 @@
...
+. Click btn:[Add Token Filter].
+. In the *Name* field, enter a name for the token filter.
+
+You can create any of the following custom token filters: 
+
+* <<dict-compound,dict_compound>>: Use a wordlist to find and create tokens from compound words in existing tokens.
+* <<edge-ngram,edge_ngram>>: Use a set character length to create tokens from the start or end of existing tokens.
+* <<elision,elision>>: Use a wordlist to remove elisions from input tokens.
+* <<keyword-marker,keyword_marker>>: Use a wordlist of keywords to find and create new tokens.
+* <<length,length>>: Use a set character length to filter out tokens that are too long or too short.
```
Comment (osfameron) [inline modules/search/pages/create-custom-token-filter.adoc:35]: if the wording is identical you don't need the item after the comma do you?

File: modules/search/pages/create-custom-token-filter.adoc [hunk 19]
```diff
@@ -0,0 +1,194 @@
...
+* <<ngram,ngram>>: Use a set character length to create new tokens.
+* <<normalize-unicode,normalize_unicode>>: Use Unicode Normalization to convert tokens.
+* <<shingle,shingle>>: Use a set character length and separator to concatenate and create new tokens.
+* <<stop-tokens,stop_tokens>>: Use a wordlist to find and remove words from tokens.
+* <<truncate-token,truncate_token>>: Use a set character length to truncate existing tokens.
+
+[#dict-compound]
+=== Create a Custom `dict_compound` Token Filter
+
+include::partial$custom-token-filters-descriptions.adoc[lines=2..3;9..27]
```
Comment (osfameron) [inline modules/search/pages/create-custom-token-filter.adoc]: huh? why are we hardcoding lines here? We can tag them.

File: modules/search/pages/create-custom-token-filter.adoc [hunk 20]
```diff
@@ -0,0 +1,194 @@
...
+. In the *Type* field, select *dict_compound*. 
+. In the *Sub Words* list, select the wordlist to use to find subwords in input tokens.
++
+You can choose your own xref:create-custom-wordlist.adoc[custom wordlist] or a xref:default-wordlists-reference.adoc[default wordlist]. Each subword match creates a new token. 
+. Click btn:[Save]. 
+
+[#edge-ngram]
+=== Create a Custom `edge_ngram` Token Filter 
+
+include::partial$custom-token-filters-descriptions.adoc[lines=31..32;38..56]
```
Comment (osfameron) [inline modules/search/pages/create-custom-token-filter.adoc]: Yeah, strongly dislike this lines syntax. We can do with tags/exclusions, happy to discuss!

File: modules/search/pages/create-search-index-rest-api.adoc [hunk 21]
```diff
@@ -0,0 +1,55 @@
...
+
+* You've installed the Couchbase command-line tool (CLI).
+
+* You have the hostname or IP address for your database. 
+
+== Procedure 
+
+To create a Search index with the REST API: 
+
+. In the command-line tool, start a `curl` command with the `XPUT` verb.
```
Comment (osfameron) [inline modules/search/pages/create-search-index-rest-api.adoc]: "In the command-line tool, start a `curl` command" sounds odd.

Comment (osfameron) [reply modules/search/pages/create-search-index-rest-api.adoc]: As in, I don't know what it means. Oh, "the Couchbase command-line tool (CLI)" above... but no wiser.
Is that `cbshell`?
But you can run `curl` from your terminal of choice.

Or does "start" have a technical meaning in this specific CLI?

File: modules/search/pages/customize-index.adoc [hunk 22]
```diff
@@ -0,0 +1,138 @@
...
+[#custom-filters]
+== Custom Filters 
+
+Custom filters are components of a Search index <<analyzers,analyzer>>. 
+
+Create and add these components to a custom analyzer to improve search results and performance for an index. 
+
+You can create the following custom filters: 
+
+* <<character-filters,>>
```
Comment (osfameron) [inline modules/search/pages/customize-index.adoc:83]: can you just omit the comma here?

File: modules/search/pages/default-analyzers-reference.adoc [hunk 23]
```diff
@@ -0,0 +1,93 @@
...
+
+The following default analyzer options are available: 
+
+[cols="1,2"]
+|====
+|Analyzer |Description 
+
+|inherit | If you set an analyzer to `inherit`, the Search index component inherits the xref:set-advanced-settings.adoc#default-analyzer[default analyzer set for an index]. 
+
+|ar |The `ar` analyzer uses character filters, tokenizers, and token filters designed for Arabic language searches.
```
Comment (osfameron) [inline modules/search/pages/default-analyzers-reference.adoc:25]: Would this be clearer as just "Arabic"?
e.g. remove all the long duplicated text to focus on the important changing part.

File: modules/search/pages/default-analyzers-reference.adoc [hunk 24]
```diff
@@ -0,0 +1,93 @@
...
+
+|hi |The `hi` analyzer uses character filters, tokenizers, and token filters designed for Hindi language searches.
+
+|hr |The `hr` analyzer uses character filters, tokenizers, and token filters designed for Croatian language searches.
+
+|hu |The `hu` analyzer uses character filters, tokenizers, and token filters designed for Hungarian language searches.
+
+|it |The `it` analyzer uses character filters, tokenizers, and token filters designed for Italian language searches.
+
+|keyword a|
```
Comment (osfameron) [inline modules/search/pages/default-analyzers-reference.adoc:55]: might be clearer with the special analyzers all listed above, and then a list of the language codes separately?

File: modules/search/pages/default-token-filters-reference.adoc [hunk 25]
```diff
@@ -0,0 +1,213 @@
...
+
+For more information about token filters, see xref:customize-index.adoc#token-filters[Token Filters].
+
+The following token filters are available: 
+
+[cols="1,2"]
+|====
+|Token Filter Type |Description 
+
+|apostrophe | Removes all characters after an apostrophe (') from tokenizer results. Also removes the apostrophe.
```
Comment (osfameron) [inline modules/search/pages/default-token-filters-reference.adoc]: add an example?

Comment (osfameron) [inline modules/search/pages/default-token-filters-reference.adoc]: That single `'` has confused github's syntax highlighter (e.g. rest of this page is italicised as it thinks it's part of the same string... 🤦 
Please double-check that the actual Asciidoc output looks fine?

File: modules/search/pages/field-data-types-reference.adoc [hunk 26]
```diff
@@ -0,0 +1,36 @@
...
+|disabled | This field data type is deprecated. It's included for compatibility only. 
+
+|geopoint a|
+
+The field contains geopoint (latitude and longitude) data, represented as either: 
+
+* A string, as two numeric values separated by a comma. 
+* A string, as a geohash point. 
+* An array, as two floating point integers. 
+* A JSON object, as the properties `"lon"`/`"lng"` and `"lat"`.
```
Comment (osfameron) [inline modules/search/pages/field-data-types-reference.adoc]: "*as* the properties" ?

you don't need the string quotes for the properties do you?

File: modules/search/pages/quick-index-field-options.adoc [hunk 27]
```diff
@@ -0,0 +1,69 @@
...
+
+The Search Service automatically applies an xref:customize-index.adoc#analyzers[analyzer] to the field's contents based on the selected language. 
+
+For more information about the available language options, see xref:quick-index-supported-languages.adoc[].
+
+|Include in search results a|
+
+To include content from the field in search results, select *Include in search results*. 
+
+To exclude the field's content from search results, clear *Include in search results*.
```
Comment (osfameron) [inline modules/search/pages/quick-index-field-options.adoc:33]: I'm finding these oppositional pairs of sentences hard to read, as they look almost identical.

File: modules/search/pages/search-index-params.adoc [hunk 28]
```diff
@@ -0,0 +1,1004 @@
...
+Your JSON payload must contain the properties described in <<initial,>>, including the <<params,>>.
+
+[#initial]
+== Initial Settings 
+
+The start of the JSON payload for a Search index contains important settings for your index: 
+
+[source,json]
+----
+include::example$simple-search-index-payload.jsonc[lines=1..14]
```
Comment (osfameron) [inline modules/search/pages/search-index-params.adoc]: EEEK PLEASE GOD NO. I liked what you did in a different .jsonc with tags much more.

File: modules/search/pages/search-index-params.adoc [hunk 29]
```diff
@@ -0,0 +1,1004 @@
...
+
+[cols="1,1,1,2"]
+|====
+|Property |Type |Required? |Description
+
+|\{char_filter_name\} |Object |Yes a|
+
+Set the name of this object to the name you want for your custom character filter. 
+
+You can reference the `\{char_filter_name\}` object elsewhere in your Search index definition to use the character filter.
```
Comment (osfameron) [inline modules/search/pages/search-index-params.adoc:327]: do you need the escapes inside backticks?

File: modules/search/partials/custom-token-filters-descriptions.adoc [hunk 30]
```diff
@@ -0,0 +1,255 @@
+// tag::dict[]
+A `dict_compound` token filter uses a wordlist to find subwords inside an input token. 
+If the token filter finds a subword inside a compound word, it turns it into a separate token. 
+
+[source,json]
+----
+include::example$complex-search-index-payload.jsonc[lines=59..62]
```
Comment (osfameron) [inline modules/search/partials/custom-token-filters-descriptions.adoc]: tag

File: modules/search/partials/nav.adoc [hunk 31]
```diff
@@ -0,0 +1,33 @@
+  ** xref:7.2@server:search:search.adoc[]
```
Comment (osfameron) [inline modules/search/partials/nav.adoc:1]: This is *very deeply nested* compared to many of our other navs - 

I found it a little hard to navigate but my use-case was "Docs Code Reviewer" 
(e.g. I wanted to see where an Asciidoc I was looking at was output in the site, and it was hard to discover it)
so this might be find for actual users?

**Review by osfameron — CHANGES_REQUESTED** [review]
Big PR, impressive work!
Overall amazing improvement, and most of my nitpicks inline can be improved iteratively.

Marking "Request changes" only because the includes with `[lines=...]` frighten me, as they will go out of date in a confusing and impossible-to-debug way.

That's a blunt instrument that we could use judiciously for code files that we don't control.

File: modules/search/partials/nav.adoc (hunk 31 above)
Comment (sarahlwelton) [reply modules/search/partials/nav.adoc:1]: Very deeply nested for good reason, because of where it needs to appear in the nav in Search (under developer guides). Take a look when it's back on Staging and you can see how it looks not just in code :)

File: modules/search/examples/autocomplete-sample.js (hunk 2 above)
Comment (sarahlwelton) [reply modules/search/examples/autocomplete-sample.js:26]: Can you show me what you mean? I didn't write this code sample.

File: modules/search/examples/create-search-index-header.sh (hunk 3 above)
Comment (sarahlwelton) [reply modules/search/examples/create-search-index-header.sh]: Do you mean an indent needs to be added or that it needs to be removed?

File: modules/search/examples/run-search-full-request.jsonc (hunk 5 above)
Comment (sarahlwelton) [reply modules/search/examples/run-search-full-request.jsonc:5]: Would be nice, yeah - maybe a note for future improvements?

File: modules/search/pages/create-child-field.adoc (hunk 10 above)
Comment (sarahlwelton) [reply modules/search/pages/create-child-field.adoc:41]: Looks weird how? Trust me, it's how it appears in the UI as an option, so that's unfortunately what we need to go with.... Very unfortunately.

File: modules/search/pages/create-child-field.adoc (hunk 9 above)
Comment (sarahlwelton) [reply modules/search/pages/create-child-field.adoc:29]: I'd be open to "Expand *Customize Index > Mappings*". Let's try that

File: modules/search/pages/create-child-field.adoc (hunk 7 above)
Comment (sarahlwelton) [reply modules/search/pages/create-child-field.adoc]: The title is a verb phrase, so it appears as "If the field contains a JSON object, Create a Child Mapping, instead."

File: modules/search/pages/create-child-field.adoc (hunk 12 above)
Comment (sarahlwelton) [reply modules/search/pages/create-child-field.adoc:53]: For displaying nicely in the table, I liked having them on separate lines.

File: modules/search/pages/create-child-field.adoc (hunk 14 above)
Comment (sarahlwelton) [reply modules/search/pages/create-child-field.adoc:70]: That's not how it appears in the UI. Trying to match the case and appearance of the options as they're shown in the UI.

File: modules/search/pages/create-child-mapping.adoc (hunk 15 above)
Comment (sarahlwelton) [reply modules/search/pages/create-child-mapping.adoc:16]: Parallelism with the previous lines :) That's all

File: modules/search/pages/create-child-mapping.adoc (hunk 16 above)
Comment (sarahlwelton) [reply modules/search/pages/create-child-mapping.adoc:28]: The field is literally called {}. I didn't notice any issues in build.

File: modules/search/pages/create-custom-date-time-parser.adoc (hunk 17 above)
Comment (sarahlwelton) [reply modules/search/pages/create-custom-date-time-parser.adoc]: I don't know... Hmm. Let me do a think.

File: modules/search/examples/autocomplete-sample.js (hunk 2 above)
Comment (osfameron) [reply modules/search/examples/autocomplete-sample.js:26]:
```
`Input <${testStr}>, len: ${testStr.length}`
```

(This is a niggle, don't worry about it, it's just I noticed some of the other language samples did it with equivalent mechanisms, and I wondered why not in JS)

File: modules/search/pages/create-custom-token-filter.adoc (hunk 18 above)
Comment (sarahlwelton) [reply modules/search/pages/create-custom-token-filter.adoc:35]: I'm just keeping it consistent with the other tags that I added

File: modules/search/pages/create-custom-token-filter.adoc (hunk 19 above)
Comment (sarahlwelton) [reply modules/search/pages/create-custom-token-filter.adoc]: When I was doing local builds with the tags, the behaviour wasn't as expected. I can elaborate further if you want to discuss, but trust me, it was the only option that made sense.

File: modules/search/pages/create-search-index-rest-api.adoc (hunk 21 above)
Comment (sarahlwelton) [reply modules/search/pages/create-search-index-rest-api.adoc]: I think I stole this from existing documentation, if you have a suggestion for how to improve the verbiage, I am all ears :)

File: modules/search/pages/customize-index.adoc (hunk 22 above)
Comment (sarahlwelton) [reply modules/search/pages/customize-index.adoc:83]: Does it work without the comma?

File: modules/search/examples/create-search-index-header.sh (hunk 3 above)
Comment (osfameron) [reply modules/search/examples/create-search-index-header.sh]: So, I would write:

```
curl -s .... \
  -u ... \
  -whatever
```

e.g. push the flags that are on a new line so they are more clearly *contained* within the same `curl` command.
Bash doesn't really have a style-guide per se, but e.g. https://lug.fh-swf.de/vim/vim-bash/StyleGuideShell.en.pdf section 7 does seem to agree with indenting here.

I've never seen anyone put options after a continuation at column 0, but it seems to happen in a few of our examples here...
My guess is someone did it once, and this example is a copy/paste.

Either that, or someone has oddly configured editor settings?

File: modules/search/pages/default-analyzers-reference.adoc (hunk 23 above)
Comment (sarahlwelton) [reply modules/search/pages/default-analyzers-reference.adoc:25]: But it doesn't appear as "Arabic" in the UI - and I want to make it clear what the analyzer is doing. If someone isn't reading the whole page, I want them to be able to quickly CTRL + F and still get the information they need.

File: modules/search/pages/default-analyzers-reference.adoc (hunk 24 above)
Comment (sarahlwelton) [reply modules/search/pages/default-analyzers-reference.adoc:55]: It's in alphabetical order and matches the order in the UI.

File: modules/search/examples/run-search-full-request.jsonc (hunk 5 above)
Comment (osfameron) [reply modules/search/examples/run-search-full-request.jsonc:5]: I'll have a play... I think it should be "easy enough"...

File: modules/search/pages/default-token-filters-reference.adoc (hunk 25 above)
Comment (sarahlwelton) [reply modules/search/pages/default-token-filters-reference.adoc]: Verified many times, never noticed an issue.

File: modules/search/pages/quick-index-field-options.adoc (hunk 27 above)
Comment (sarahlwelton) [reply modules/search/pages/quick-index-field-options.adoc:33]: That's why I tried to change the order of the words in each sentence.

File: modules/search/pages/search-index-params.adoc (hunk 29 above)
Comment (sarahlwelton) [reply modules/search/pages/search-index-params.adoc:327]: It wasn't displaying properly otherwise.

File: modules/search/pages/create-child-field.adoc (hunk 7 above)
Comment (osfameron) [reply modules/search/pages/create-child-field.adoc]: Ah, I thought that was possible, but couldn't tell from the local context.

If the upstream wording ever changes to e.g. "Creating a Child Mapping" this sentence would then be mangled, so I'd prefer to add the explicit link text in this case.

File: modules/search/pages/search-index-params.adoc (hunk 28 above)
Comment (sarahlwelton) [reply modules/search/pages/search-index-params.adoc]: Again, there is a method to my madness. It wasn't displaying properly without the line includes :(

File: modules/search/pages/create-child-field.adoc (hunk 10 above)
Comment (osfameron) [reply modules/search/pages/create-child-field.adoc:41]: OK, it probably makes sense in context. I think after seeing the `a|` I was thinking "Searchable A's?!" and found it hard to parse.

File: modules/search/pages/create-child-mapping.adoc (hunk 15 above)
Comment (osfameron) [reply modules/search/pages/create-child-mapping.adoc:16]: Yeah, I think it's *not* quite parallel because "You have done X" in those cases is a one-and-done thing and logging in isn't.

File: modules/search/pages/create-search-index-rest-api.adoc (hunk 21 above)
Comment (osfameron) [reply modules/search/pages/create-search-index-rest-api.adoc]: OK, something like "Run the `curl` command with the `XPUT` verb." would be ok.
or just "Run `curl`" as you have an example following.

File: modules/search/pages/customize-index.adoc (hunk 22 above)
Comment (osfameron) [reply modules/search/pages/customize-index.adoc:83]: yeah, I think so. Not important either way though :)

File: modules/search/pages/default-analyzers-reference.adoc (hunk 23 above)
Comment (osfameron) [reply modules/search/pages/default-analyzers-reference.adoc:25]: OK 👍

File: modules/search/pages/create-custom-token-filter.adoc (hunk 19 above)
Comment (osfameron) [reply modules/search/pages/create-custom-token-filter.adoc]: Yeah, let's discuss, tagging an *adoc* should Just Work, I'd rather understand and ideally fix the issue!

File: modules/search/pages/create-custom-date-time-parser.adoc (hunk 17 above)
Comment (osfameron) [reply modules/search/pages/create-custom-date-time-parser.adoc]: OK, not important (it's a bit fragile, but I'm not as worried by this as the `include::` comments, so feel free to close this comment if you don't find a much better way 👍)

File: modules/search/examples/autocomplete-sample.js (hunk 2 above)
Comment (sarahlwelton) [reply modules/search/examples/autocomplete-sample.js:26]: If it's fine as is, then maybe leave it? lol

File: modules/search/examples/create-search-index-header.sh (hunk 3 above)
Comment (sarahlwelton) [reply modules/search/examples/create-search-index-header.sh]: Will indent, this was totally a copy/paste / stolen from existing doc.

File: modules/search/pages/quick-index-field-options.adoc (hunk 27 above)
Comment (osfameron) [reply modules/search/pages/quick-index-field-options.adoc:33]: OK, take it as a data-point of feedback to consider iterating on / leaving as is as you prefer 👍

File: modules/search/pages/create-child-mapping.adoc (hunk 15 above)
Comment (sarahlwelton) [reply modules/search/pages/create-child-mapping.adoc:16]: Well, logging in is a one-and-done isn't it? It's enter your username and password. It's something you have done.

File: modules/search/examples/autocomplete-sample.js (hunk 2 above)
Comment (osfameron) [reply modules/search/examples/autocomplete-sample.js:26]: Yup!

File: modules/search/pages/create-custom-token-filter.adoc [hunk 32]
```diff
@@ -42,7 +42,9 @@ You can create any of the following custom token filters:
 [#dict-compound]
 === Create a Custom `dict_compound` Token Filter
 
-include::partial$custom-token-filters-descriptions.adoc[lines=2..3;9..27]
+include::partial$custom-token-filters-descriptions.adoc[tags=dict;!dict_example]
+
+// include::partial$custom-token-filters-descriptions.adoc[lines=2..3;9..27]
```
Comment (osfameron) [inline modules/search/pages/create-custom-token-filter.adoc]: Probably tidier to remove these comments now.

**Review by osfameron — APPROVED** [review]
Thanks, looks good! 👍
